corva/application.py

```diff
@@ -1,2 +1,64 @@
+from typing import Any, Callable, List, Optional
+
+from corva.models.stream import StreamContext, StreamEvent
+from corva.network.api import Api
+from corva.settings import CorvaSettings, CORVA_SETTINGS
+from corva.stream import stream_runner
+
+
 class Corva:
-    pass
+    def __init__(
+         self,
+         api_url: Optional[str] = None,
+         data_api_url: Optional[str] = None,
+         cache_url: Optional[str] = None,
+         api_key: Optional[str] = None,
+         app_key: Optional[str] = None,
+         api_timeout: Optional[int] = None,
```

Comment:
I think since this is an external interface, we should have better names here. Here are my preferences:

* remove data_api_url: I'm also not sure that we need to be able to override the data API URL here.
* api_timeout => timeout
* api_max_retries => max_retries
* cache_kwargs => cache_settings

Reply:
agreed
might as well delete, cache_url and api_url along with data_api_url

Reply:
agreed
might as well delete all params that can be retreived from SETTINGS variable along with data_api_url
so we will be left with timeout, max_retries and cache_settings